packages/client/src/schema/repository.ts

```diff
@@ -1544,6 +1546,35 @@ export class RestRepository<Record extends XataRecord>
     }
   }
 
+  async #deleteRecords(recordIds: string[], columns: SelectableColumn<Record>[] = ['*']) {
+    const fetchProps = await this.#getFetchProps();
+
+    const operations = recordIds.map((id) => ({ delete: { table: this.#table, id } }));
+
+    try {
+      // TODO: Add support for columns in transaction bulk delete
+      const objects = await this.read(recordIds, columns);
+
+      await branchTransaction({
+        pathParams: {
+          workspace: '{workspaceId}',
+          dbBranchName: '{dbBranch}',
+          region: '{region}'
+        },
+        body: { operations },
+        ...fetchProps
+      });
+
+      return objects;
+    } catch (e) {
+      if (isObject(e) && e.status === 404) {
+        return null;
+      }
```

Comment:
This needs to be updated to the new error the backend returns (in delete might never happen), will add a special type guard to extract transactional errors.